### PR TITLE
Frost walker

### PR DESCRIPTION
**Pull request by CoolMagic233**

### CODE REVIEW

**Comment by Mcayear** [comment]
linked #257

**Comment by Mcayear** [comment]
ice will not melt.
冰不会消融。
image

**Comment by Mcayear** [comment]
When staying on the water surface, ice should also melt.

vanilla:
image

**Comment by CoolMagic233** [comment]
> When staying on the water surface, ice should also melt.
> 
> vanilla: image

File: src/main/java/cn/nukkit/Player.java [hunk 1]
```diff
@@ -2081,6 +2085,34 @@ protected void handleMovement(Vector3 clientPos) {
...
+            //Handle frost walker enchantment
+            Enchantment frostWalker = inventory.getBoots().getEnchantment(Enchantment.ID_FROST_WALKER);
+            if (frostWalker != null && frostWalker.getLevel() > 0 && !this.isSpectator() && this.y >= 1 && this.y <= 255) {
+                int radius = 2 + frostWalker.getLevel();
+                for (int coordX = this.getFloorX() - radius; coordX < this.getFloorX() + radius + 1; coordX++) {
+                    for (int coordZ = this.getFloorZ() - radius; coordZ < this.getFloorZ() + radius + 1; coordZ++) {
+                        Block up = level.getBlock(coordX, this.getFloorY(), coordZ);
+                        Block block = level.getBlock(coordX, this.getFloorY() - 1, coordZ);
+                        if ((block.getId() == Block.STILL_WATER || block.getId() == Block.WATER) && up.isAir()) {
+                            if (block instanceof BlockLiquid liquid){
```
Comment (lt-name) [inline src/main/java/cn/nukkit/Player.java:2099]: 感觉可以直接把`block.getId() == Block.STILL_WATER || block.getId() == Block.WATER`改成`block instanceof BlockWater`

Comment (CoolMagic233) [reply src/main/java/cn/nukkit/Player.java:2099]: 岩浆也算流体啊

**Comment by Mcayear** [comment]
测试了没问题，合并吧

**Review by lt-name — APPROVED** [review]

**Pull request merged**